llvm/include/llvm/ProfileData/MemProf.h

```diff
@@ -491,11 +492,15 @@ struct MemProfRecord {
   }
 };
 
+// A "typedef" for GUID.  See ScalarTraits<memprof::GUIDHex64> for how a GUID is
+// serialized and deserialized in YAML.
+LLVM_YAML_STRONG_TYPEDEF(uint64_t, GUIDHex64)
```

Comment:
Is there enough YAML specific stuff in the header to make factoring it out into it's own header worthwhile?

Reply:
Yes, let me create `MemprofYAML.h` or something in a subsequent patch.